testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1834,6 +1844,69 @@ public void testCreateTable()
         assertFalse(getQueryRunner().tableExists(getSession(), tableNameLike));
     }
 
+    @DataProvider
+    public Object[][] likeTableParentTableCandidates()
+            throws Exception
+    {
+        return likeTableParentTableCandidatesData();
+    }
+
+    protected Object[][] likeTableParentTableCandidatesData()
+            throws Exception
+    {
+        String tableName = "test_create_orig_" + randomTableSuffix();
+        return new Object[][] {
+                {tableName, "CREATE TABLE " + tableName + " (a bigint, b double, c varchar(50))"}
+        };
+    }
+
+    @Test(dataProvider = "likeTableParentTableCandidates")
+    public void testCreateTableLikeIncludingProperties(String tableName, String createTableSql)
+    {
+        String catalogName = getSession().getCatalog().orElseThrow();
+        assertUpdate(createTableSql);
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+        assertTableColumnNames(tableName, "a", "b", "c");
+
+        String tableNameLike = "test_create_like_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableNameLike + " (LIKE " + tableName + " INCLUDING PROPERTIES, d bigint, e varchar(50))");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableNameLike));
+        assertTableColumnNames(tableNameLike, "a", "b", "c", "d", "e");
+
+        String parentTableProperties = getTablePropertiesString(tableName);
```

Comment:
FYI Hive & Iceberg offer a rather simpler way to retrieve the property values:

```
        assertThat(onTrino().executeQuery("SELECT key, value FROM \"mytable$properties\"" + trinoPropertiesTableName))
                .containsOnly(
                        row("custom.table-property", "my_custom_value"),
                        row("foo", "bar"));
```

I find `getTablePropertiesString` with the regex matching is rather unstable.

Reply:
This code is however in the BaseConnectorTest. So might be applicable to other connectors not supporting $properties metadata. The regex should be stable as long ignored property values don't contain a comma: `" = (.*),?"`. I would think that's probably not so common.

Reply:
I would suggest having a corresponding test only in the relevant test classes for the connectors that actually deal with table properties (e.g. : hive, delta, iceberg, ?clickhouse?). There is no need to  test a connector for this functionality if it doesn't have any support for table properties.

Reply:
> @mdesmet wondering, is it possible and worth to add a test-case in the [BaseConnectorTest](https://github.com/trinodb/trino/blob/master/testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java#L126)

@osscm : as you mentioned adding a test case in [BaseConnectorTest]. What's your view on this? It works automatically without having to write extra tests for delta and iceberg now. Is a more explicit connector specific test a better approach?